### PR TITLE
Fix 3494 deprecation upgrade bug on master

### PR DESCRIPTION
**Pull request by MetRonnie**
Replica of PR #3524 from 7.8.x, but with extra change to deal with `[scheduling][dependencies][X]graph -> [scheduling][graph]X` which is handled separately

### CODE REVIEW

File: tests/deprecations/02-overwrite.t [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test that upgraded deprecations don't overwrite existing items
+. $(dirname $0)/test_header
```
Comment (wxtim) [inline tests/deprecations/02-overwrite.t:19]:
```suggestion
. "$(dirname $0)/test_header"
```

File: tests/deprecations/02-overwrite.t [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#-------------------------------------------------------------------------------
+set_test_number 1
+#-------------------------------------------------------------------------------
+install_suite $TEST_NAME_BASE $TEST_NAME_BASE
```
Comment (wxtim) [inline tests/deprecations/02-overwrite.t:23]:
```suggestion
install_suite "$TEST_NAME_BASE" "$TEST_NAME_BASE"
```

File: tests/deprecations/02-overwrite.t [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#-------------------------------------------------------------------------------
+cylc validate $SUITE_NAME 2> val.out
```
Comment (wxtim) [inline tests/deprecations/02-overwrite.t:25]:
```suggestion
cylc validate "$SUITE_NAME" 2> val.out
```

File: tests/deprecations/02-overwrite.t [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+TEST_NAME=${TEST_NAME_BASE}-original-events-not-overwritten
+grep_ok "UpgradeError" "val.out"
+#-------------------------------------------------------------------------------
+purge_suite $SUITE_NAME
```
Comment (wxtim) [inline tests/deprecations/02-overwrite.t:29]:
```suggestion
purge_suite "$SUITE_NAME"
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -69,6 +69,7 @@ requests_).
  - Dima Veselov
  - Gilliano Menezes
  - Mel Hall
+ - Ronnie Dutta
```
Comment (wxtim) [inline CONTRIBUTING.md:72]: :+1:

File: tests/deprecations/02-overwrite.t [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (wxtim) [inline tests/deprecations/02-overwrite.t:1]: command to replicated shellchecker on your desktop
```bash
cd my-cylc-dir
.travis/shellchecker
```

File: tests/deprecations/02-overwrite.t [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+#-------------------------------------------------------------------------------
+# Test that upgraded deprecations don't overwrite existing items
+. $(dirname $0)/test_header
+#-------------------------------------------------------------------------------
+set_test_number 1
+#-------------------------------------------------------------------------------
+install_suite $TEST_NAME_BASE $TEST_NAME_BASE
+#-------------------------------------------------------------------------------
+cylc validate $SUITE_NAME 2> val.out
+TEST_NAME=${TEST_NAME_BASE}-original-events-not-overwritten
```
Comment (wxtim) [inline tests/deprecations/02-overwrite.t:26]: You can remove this line entirely, after all, there's only one test in this file. (I like this - it's easier to debug when the tests are granular)
```suggestion
```

**Comment by wxtim** [comment]
:+1:  It's looking pretty good so far - you've got some tests to fix - I wrote them, so feel free to ask if you want more info.

**Comment by MetRonnie** [comment]
Can anyone explain what's gone wrong with the Travis unit test?

**Comment by wxtim** [comment]
This file should see you right.
[patch.txt](https://github.com/cylc/cylc-flow/files/4426775/patch.txt)

**Comment by MetRonnie** [comment]
Actually it was due to my edits - not `except`ing a `KeyError` when a suite doesn't contain a `[scheduling]` section.

File: cylc/flow/cfgspec/suite.py [hunk 8]
```diff
@@ -323,28 +324,37 @@ def upg(cfg, descr):
...
-                u.show_keys(['scheduling', 'dependencies', 'X', 'graph']),
-                u.show_keys(['scheduling', 'graph', 'X']),
-                '\n'.join(sorted(keys)),
-            )
-
+    # Upgrader cannot do this type of move:
+    try:  # Upgrade cfg['scheduling']['dependencies']['graph']
+        if 'dependencies' in cfg['scheduling']:
+            msgOld = '[scheduling][dependencies][X]graph'
+            msgNew = '[scheduling][graph]X'
```
Comment (hjoliver) [inline cylc/flow/cfgspec/suite.py:331]: I know this seems kind of trivial but camelCase is considered a style violation in Python https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

So, could you change this to `msg_old` etc.?

Comment (MetRonnie) [reply cylc/flow/cfgspec/suite.py]: Sorry, JavaScript habit!

**Pull request merged**